File: user-service/src/main/java/rs/raf/user_service/repository/CompanyRepository.java
package rs.raf.user_service.repository;

import org.springframework.data.jpa.repository.JpaRepository;


import rs.raf.user_service.domain.entity.Company;

import java.util.List;
import java.util.Optional;

public interface CompanyRepository extends JpaRepository<Company, Long> {
    Optional<Company> findByRegistrationNumber(String registrationNumber);

    List<Company> findByMajorityOwner_Id(Long id);

    Optional<Company> findByTaxId(String taxId);

}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/repository/CompanyRepository.kt
package pack.userservicekotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pack.userservicekotlin.domain.entities.Company
import java.util.*

@Repository
interface CompanyRepository : JpaRepository<Company, Long> {
    fun findByRegistrationNumber(registrationNumber: String): Optional<Company>

    // bio kao _Id
    fun findByMajorityOwnerId(id: Long): List<Company>

    fun findByTaxId(taxId: String): Optional<Company>
}
